pkg/registry/core/service/storage/rest.go

```diff
@@ -559,8 +548,14 @@ func (r *REST) getAllocatorByClusterIP(service *api.Service) ipallocator.Interfa
 	}
 
 	secondaryAllocatorCIDR := r.secondaryServiceIPs.CIDR()
-	if netutil.IsIPv6String(service.Spec.ClusterIP) && netutil.IsIPv6CIDR(&secondaryAllocatorCIDR) {
-		return r.secondaryServiceIPs
+	if netutil.IsIPv6CIDR(&secondaryAllocatorCIDR) {
+		if netutil.IsIPv6String(service.Spec.ClusterIP) {
+			return r.secondaryServiceIPs
+		}
+	} else {
+		if !netutil.IsIPv6String(service.Spec.ClusterIP) {
+			return r.secondaryServiceIPs
```

Comment:
I think this is a second bugfix not mentioned in the PR summary / release note: previously, it required that in a dual-stack cluster, the IPv4 addresses be primary and the IPv6 addresses be secondary, but with this change either order will work?

Reply:
That's correct, I will clarify that in the commit and release note.

Reply:
(tests already verify that)